pkg/controller/main-controller.go

```diff
@@ -356,8 +358,13 @@ func (c *Controller) startUpgradeServer(ctx context.Context) <-chan error {
 // startUpgradeServer Starts the Upgrade tenant API server and notifies the start and stop via notificationChannel
 func (c *Controller) startSTSAPIServer(ctx context.Context, notificationChannel chan<- *EventNotification) {
 	klog.Infof("Starting STS API server")
-	publicCertPath, publicKeyPath := c.waitSTSTLSCert()
-	certsManager, err := xcerts.NewManager(ctx, *publicCertPath, *publicKeyPath, LoadX509KeyPair)
+
+	publicCertPath := miniov2.GetPublicCertFilePath("sts")
+	privateKeyPath := miniov2.GetPrivateKeyFilePath("sts")
+	if utils.GetOperatorRuntime() != common.OperatorRuntimeOpenshift {
+		c.waitSTSTLSCert()
```

Comment:
That should be `publicCertPath, publicKeyPath = c.waitSTSTLSCert()`?

Reply:
Agree, it should be overwritten here as `publicCertPath, privateKeyPath = c.waitSTSTLSCert()`, right?

Reply:
in the end the `GetPublicCertFilePath("sts")` returns the same files path and overwrite it would end in the same value, but it harms nothing make it explicit.
Applied suggestion